[Kommander] custom domains and certificates for managed clusters (#4553)
mesosphere/dcos-docs-site: pages/dkp/kommander/2.3/clusters/custom-domain-certificate/index.md

---
layout: layout.pug
navigationTitle: Custom domain and certificate
title: Configure a custom domain and certificate
menuWeight: 30
excerpt: Configure a custom domain and certificate for your Management or any Managed/Attached cluster
beta: false
enterprise: false
---

> DKP supports configuring a custom domain name for accessing the DKP UI and other platform services. Additionally, you can provide a custom certificate for each domain, or one can be issued automatically by Let's Encrypt, or other certificate authorities supporting the ACME protocol.

Reviewer: `you can provide a custom certificate for each domain`

There is only 1 domain per cluster. So we either need to change this to `for a domain` or explain that custom domain can be configured per cluster.

Author: ```suggestion
DKP supports configuring a custom domain name per cluster for accessing the DKP UI and other platform services. Additionally, you can provide a custom certificate for each domain, or one can be issued automatically by Let's Encrypt, or other certificate authorities supporting the ACME protocol.
```

> The configuration path is different depending on the cluster for which you want to customize the domain or certificate.

Author: ```suggestion
The configuration path is the same regardless of whether you are configuring a custom domain and certificate on the management, or a managed/attached cluster.
```

[Customize a domain or certificate in the **Management Cluster**][management] (during installation).

> [Customize a domain or certificate in a **Managed Cluster**][managed] (after installation).

Author: ```suggestion
[Customize a domain or certificate in the [**Management or a Managed/Attached Cluster**][managed] (after installation).
```

> [Customize a domain or certificate in a **Managed Cluster**][managed] (after installation).

Author: Addressed in https://github.com/mesosphere/dcos-docs-site/pull/4553/commits/e4c6941f6078771b4a35480f70238368e049bad7

> ## Why to customize domains

Author: ```suggestion
## Reasons for customizing your DNS domain
```

> DKP’s default configuration supports the customization of domains to allow you to use your own domain or hostname for your services.

Reviewer: I wouldn't say that "default configuration supports" something, but rather something like "DKP supports the customization of domain ... ". By default, we don't install DKP with custom domain.

Author: ```suggestion
DKP supports the customization of domains to allow you to use your own domain or hostname for your services.
```

> For example, you can set up your DKP UI to be accessible via the provided domain instead of the URL provided by default. You can also use your domain to enable your company to access back end services, or your customers to access front end services via a specific domain. Or you can configure a custom domain to make a cluster accessible with a specific domain.

Reviewer: > DKP UI to be accessible via the provided **domain** instead of the **URL** provided by default.

We should use a single term for domain across the whole document.

Author: ```suggestion
For example, you can set up your DKP UI to be accessible via your custom domain name instead of the domain provided by default. You can also use your domain to enable your company to access back-end services, or your customers to access front-end services via a specific domain. Or you can configure a custom domain to make a cluster accessible via a specific domain.
```

> ## Why to customize certificates

Author: ```suggestion
## Reasons for using a custom certificate
```

> DKP’s default CA identity supports the encryption of data exchange and traffic (between your client and your environment’s server). To configure an additional security layer that validates your environment’s server authenticity, DKP supports configuring a custom certificate by a Certificate Authority with an ACME protocol like Let’s Encrypt, or a custom certificate created specifically for your organization.

Reviewer 2: ```suggestion
DKP’s default CA identity supports the encryption of data exchange and traffic (between your client and your environment’s server). To configure an additional security layer that validates your environment’s server authenticity, DKP supports configuring a custom certificate issued by a trusted Certificate Authority either directly in a Secret or managed automatically using the ACME protocol (e.g. Let’s Encrypt).
```

> Changing the default certificate for any of your clusters by modifying the ingress via an API can be helpful. For example, you can adapt it to classify your DKP UI or any other type of service as trusted (when accessing a service via a browser).

Reviewer 2: What do you mean by `by modifying the ingress via an API`? I would suggest to either remove that part or rephrase it differently

Reviewer: I would remove that part as well. This is an introduction page. We show specifics on how to configure custom domain/cert via modifying the `Ingress` spec of a `KommanderCluster` object in the following pages.

Author: ```suggestion
Changing the default certificate for any of your clusters can be helpful, for example, you can adapt it to classify your DKP UI or any other type of service as trusted (when accessing a service via a browser).
```
